.travis.yml

```diff
@@ -0,0 +1,35 @@
+notifications:
+  email: false
+
+branches:
+  only:
+    - master
+    - develop
+    - /^v\d+\.\d+\.\d+/
+
+language: node_js
+
+node_js:
+  - '10'
+  - '12'
+
+install:
+  - yarn install --frozen-lockfile --non-interactive
+
+jobs:
+  include:
+    - stage: Check TypeScript
+      script:
+        - yarn check:types
+    - stage: Lint
+      script:
+        - yarn lint
+    - stage: Check Formatting
+      script:
+        - yarn format:check
+    - stage: Test
+      script:
+        - yarn test
+    - stage: Build
+      script:
+        - yarn build
```

Comment:
I'm not too familiar with travis, do these steps run synchronously?

Reply:
Yep! Scripts in the same stage will run in parallel, but scripts in separate stages will run in sequence.

Reply:
Okay cool! That's what I thought. I wonder why travis is [seemingly running these out of sequence then](https://github.com/FormidableLabs/next-urql/pull/30/checks?check_run_id=421127927) 🤔

Reply:
Looks like we need a [`stages` key to enforce order](https://docs.travis-ci.com/user/build-stages/#specifying-stage-order-and-conditions)

Reply:
What do you think about using `name` for the human-readable stage name?

```
- stage: check_typescript
      name: "Check TypeScript"
      script:
        - yarn check:types
    ...
```

then we could enforce order:

```
stages:
  - check_typescript
  ...
```

Reply:
Ah good catch! One thing I'm realizing is that it's actually not critical for any of these stages to run in order. So while we _could_ enforce it, I'm not sure it's really worth it. I haven't seen any of our other projects do it before, but that's not necessarily a reason to _not_ do it.

cc/ @ryan-roemer are there any benefits or drawbacks to enforcing stage order in a build like this where we don't need to persist any data across stages?